framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -2283,6 +2300,35 @@ VulkanReplayConsumerBase::OverrideCreateInstance(VkResult original_result,
                     // Remove enabled extensions that are not available from the replay instance.
                     feature_util::RemoveUnsupportedExtensions(available_extensions, &filtered_extensions);
                 }
+                else if (options_.use_colorspace_fallback)
+                {
+                    if (!feature_util::IsSupportedExtension(available_extensions,
+                                                            VK_EXT_SWAPCHAIN_COLOR_SPACE_EXTENSION_NAME))
+                    {
+                        auto iter = std::find_if(
+                            filtered_extensions.begin(), filtered_extensions.end(), [](const char* extension) {
+                                return util::platform::StringCompare(VK_EXT_SWAPCHAIN_COLOR_SPACE_EXTENSION_NAME,
+                                                                     extension) == 0;
+                            });
+                        if (iter != filtered_extensions.end())
+                        {
+                            filtered_extensions.erase(iter);
+                        }
+                    }
+                    if (!feature_util::IsSupportedExtension(available_extensions,
+                                                            VK_AMD_DISPLAY_NATIVE_HDR_EXTENSION_NAME))
+                    {
+                        auto iter = std::find_if(
+                            filtered_extensions.begin(), filtered_extensions.end(), [](const char* extension) {
+                                return util::platform::StringCompare(VK_AMD_DISPLAY_NATIVE_HDR_EXTENSION_NAME,
+                                                                     extension) == 0;
+                            });
+                        if (iter != filtered_extensions.end())
+                        {
+                            filtered_extensions.erase(iter);
+                        }
+                    }
```

Comment:
This is a decent amount of duplicate code. I would strongly recommend moving the logic into a helper function, just like `feature_util::RemoveUnsupportedExtensions()`. Could be called `feature_util::RemoveSupportedExtension(filtered_extensions, "VK_EXT_<extension_name>)` or something similar.